### PR TITLE
feat(client): support push query termination in Java client

### PR DESCRIPTION
**Pull request by vcrfxia**
### Description 

^ via the `/close-query` endpoint. This PR also adds a test to check that `StreamedQueryResult` objects are properly completed when the underlying query is terminated.

### Testing done 

Unit tests.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ClientImpl.java [hunk 1]
```diff
@@ -180,6 +205,23 @@ private HttpClientRequest configureBasicAuth(final HttpClientRequest request) {
...
+    final RecordParser recordParser = RecordParser.newDelimited("\n", response);
+    final QueryResponseHandler<T> responseHandler =
+        responseHandlerSupplier.get(Vertx.currentContext(), recordParser, cf);
+
+    recordParser.handler(responseHandler::handleBodyBuffer);
+    recordParser.endHandler(responseHandler::handleBodyEnd);
+    recordParser.exceptionHandler(responseHandler::handleException);
+  }
+
+  private static void handleSuccessfulCloseQueryResponse(final CompletableFuture<Void> cf) {
```
Comment (purplefox) [inline ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ClientImpl.java:221]: These methods and the one above seem like overkill - I'd probably just put them in the lambda.

File: ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ClientImpl.java [hunk 2]
```diff
@@ -158,15 +188,10 @@ private HttpClientRequest configureBasicAuth(final HttpClientRequest request) {
   private static <T> void handleResponse(
       final HttpClientResponse response,
       final CompletableFuture<T> cf,
```
Comment (purplefox) [inline ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ClientImpl.java:190]: Seems odd to pass in both a cf and a Handler<HttpClientResponse> to signal back both exceptionally and successful results. Can't the cf do both?

File: ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ClientImpl.java [hunk 3]
```diff
@@ -133,22 +139,46 @@ public void close() {
...
+      final String path,
+      final JsonObject requestBody,
+      final CompletableFuture<T> cf,
+      final Handler<HttpClientResponse> successfulResponseHandler) {
     HttpClientRequest request = httpClient.request(HttpMethod.POST,
         serverSocketAddress, clientOptions.getPort(), clientOptions.getHost(),
-        "/query-stream",
-        response -> handleResponse(response, cf, responseHandlerSupplier))
+        path,
+        response -> handleResponse(response, cf, successfulResponseHandler))
```
Comment (purplefox) [inline ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ClientImpl.java:176]: I don't think you need to pass in successfulResponseHandler here, you can just call cf.thenAccept(this::successfulResponseHandler) (or similar) in this method.

Comment (vcrfxia) [reply ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ClientImpl.java:176]: Not quite. The CompletableFuture will succeed even if the response status is not 200. The successful response handler is only meant to be called if the response status is 200, else the error handling logic should be called. I realize the way this is currently set up is a bit convoluted, though, so I've refactored it to be more readable.

File: ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ClientImpl.java (hunk 2 above)
Comment (vcrfxia) [reply ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ClientImpl.java:190]: Refactored this to be less convoluted.

File: ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ClientImpl.java (hunk 1 above)
Comment (vcrfxia) [reply ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ClientImpl.java:221]: As above, I've refactored this. Hopefully your concerns have been addressed indirectly.

**Pull request merged**